src/System.Linq.Expressions/src/System/Linq/Expressions/ExpressionStringBuilder.cs

```diff
@@ -682,32 +631,24 @@ protected internal override Expression VisitUnary(UnaryExpression node)
                 case ExpressionType.PreIncrementAssign:
                 case ExpressionType.Quote:
                     break;
-                case ExpressionType.TypeAs:
-                    Out(" As ");
-                    Out(node.Type.Name);
-                    Out(")");
-                    break;
-                case ExpressionType.PostIncrementAssign:
-                    Out("++");
-                    break;
-                case ExpressionType.PostDecrementAssign:
-                    Out("--");
-                    break;
-                default:
-                    Out(")");
-                    break;
+                case ExpressionType.TypeAs:              Out(" As ");
+                                                         Out(node.Type.Name);
+                                                         Out(')');               break;
+                case ExpressionType.PostIncrementAssign: Out("++");              break;
+                case ExpressionType.PostDecrementAssign: Out("--");              break;
+                default:                                 Out(')');               break;
```

Comment:
Keeping `default` here; all cases are covered above with `default` throwing for unreachable code. This should cover all cases where we emitted a trailing '(' above, with the exception of `TypeAs` which is treated specially. Please double-check this is indeed the case.